services/core-api/app/api/mines/tailings/party_appt/resources/party_appt_list.py

```diff
@@ -0,0 +1,134 @@
+
+import dateutil.parser
+
+from flask import request, current_app
+from flask_restplus import Resource
+from sqlalchemy import or_, exc as alch_exceptions
+from werkzeug.exceptions import BadRequest, InternalServerError, NotFound, Forbidden
+
+from app.api.parties.party.models.address import Address
+from app.extensions import api
+from app.api.utils.access_decorators import requires_role_view_all, requires_role_mine_edit
+from app.api.utils.resources_mixins import UserMixin
+from app.api.utils.custom_reqparser import CustomReqparser
+
+from app.api.mines.mine.models.mine import Mine
+from app.api.mines.permits.permit.models.permit import Permit
+from app.api.parties.party.models.party import Party
+from app.api.parties.party_appt.models.mine_party_appt import MinePartyAppointment
+from app.api.parties.party_appt.models.mine_party_appt_type import MinePartyAppointmentType
+from app.api.mines.tailings.models.tailings import MineTailingsStorageFacility
+from app.api.constants import PERMIT_LINKED_CONTACT_TYPES
+
+
+class MineTailingsStorageFacilityPartyAppointmentResource(Resource, UserMixin):
+    parser = CustomReqparser()
+    parser.add_argument('mine_guid', type=str, help='guid of the mine.')
+    parser.add_argument('mine_tailings_storage_facility_guid', type=str, help='guid of the tailings storage facility.')
+    parser.add_argument(
+        'party_name',
+        type=str,
+        help='Last name of the party (Person), or the Organization name (Organization).',
+        required=True)
+    parser.add_argument(
+        'phone_no', type=str, help='The primary phone number of the party. Ex: 123-123-1234', required=True)
+    parser.add_argument(
+        'last_name', type=str, help='Last name of the party, if the party is a person.')
+    parser.add_argument(
+        'first_name', type=str, help='First name of the party, if the party is a person.')
+    parser.add_argument('email', type=str, help='The primary email of the party.')
+    parser.add_argument(
+        'suite_no',
+        type=str,
+        store_missing=False,
+        help='The suite number of the party address. Ex: 123')
+    parser.add_argument(
+        'address_line_1',
+        type=str,
+        store_missing=False,
+        help='The first address line of the party address. Ex: 1234 Foo Road')
+    parser.add_argument(
+        'address_line_2',
+        type=str,
+        store_missing=False,
+        help='The second address line of the party address. Ex: 1234 Foo Road')
+    parser.add_argument(
+        'city',
+        type=str,
+        store_missing=False,
+        help='The city where the party is located. Ex: FooTown')
+    parser.add_argument(
+        'sub_division_code',
+        type=str,
+        store_missing=False,
+        help='The region code where the party is located. Ex: BC')
+    parser.add_argument(
+        'post_code',
+        type=str,
+        store_missing=False,
+        help='The postal code of the party address. Ex: A0B1C2')
```

Comment:
Actually, I just noticed we don't have an email field here.   There is one on the form.  Can we add that in?